server/api/api/utils.py

```diff
@@ -429,7 +430,11 @@ def validate_and_mask_notification_list(
         # validate notification schema
         mlrun.common.schemas.Notification(**notification_object.to_dict())
 
-        notification_object.validate_notification_params()
+        # skip the params validation if the key is "secret"
```

Comment:
Not sure I understand why. Also why not check this in the validation function?

Reply:
after a conversation with @quaark we decided that we should skip the validation when "secret" is present because it implies that the parameters are securely handled and should not be validated.